### ISSUE TITLE
Change in cljc file triggers rebuild of all .cljs sources

### ISSUE THREAD

**Issue by danskarda** [bhauman/lein-figwheel#255]
Hi,
in our project we migrated from CLJX files to CLJC. Unfortunately it affects speed of our builds in figwheel.

Setup: 
- app has many cljc files to share logic between server and client
- full build with :optimizations :none takes more than 30s.

Findings with figwheel:
- touch any .cljs file => recompilation takes ~ 2s
- touch any .cljc file (even empty one) => recompilation takes ~ 30s

Any .cljc file can have macros inside. So change in .cljc file can impact .cljs. To be correct one has either perform detailed dependency analysis or rebuild all .cljs files. 

Our project has very few macros and change in a macro is really rare. Can we turn off this feature off (full rebuild after .cljc change)?

Thank you,
D.

/cc @maio

**Comment by bhauman** [comment]
Shoot! I'm tempted to ignore macro changes in cljc files.

If you want to hack this in the short term:

You will want to comment out:
https://github.com/bhauman/clojurescript-build/blob/master/src/clojurescript_build/core.clj#L141

And return an empty [] from the surrounding function.

**Comment by bhauman** [comment]
I found a better solution in your build config you can specify `:reload-clj-files false` in your build config like so:

```clojure
{ :id "example"
  :source-paths ["src"]
  :reload-clj-files false
  :compiler {:main .... }
...}
```

**Comment by danskarda** [comment]
Sounds promising.

I tried `:reload-clj-files`. It seems to work on CLJ files but .CLJC files continue to trigger full build. It seems to me that the option turns of watching CLJ files, but CLJC processing is performed much later. It still triggers loading of CLJC file as Clojure file (and thus full cljs build). 

Could this be the case?

**Comment by bhauman** [comment]
Yeah this is strange.  Hmmm do you have `:recompile-dependents` set to true? In your build-options?

It's entirely possible that the flag isn't working properly. Or it's not making it to the build somehow.  But if the flag **is** working the cljc files should not be getting reloaded as Clojure sources at all, or triggering dependents.

I'm currently finishing up an extensive refactor/reduction of the figwheel code, I'm keeping this in mind.

**Comment by danskarda** [comment]
We do not touch `:recompile-dependents` option. So yes, we use default `true` value.

**Comment by danskarda** [comment]
The problem is not in figwheel, but in clojurescript-build. It ignores `:reload-clj-files` value. Or to be precise, it overrides it to true:

https://github.com/bhauman/clojurescript-build/blob/master/src/clojurescript_build/auto.clj#L186

When I removed the override, I was able to avoid full rebuild after touching .cljc file.

**Comment by bhauman** [comment]
OK this is cool.  The next release will no longer use clojurescript-build I will make sure that this flag works.

**Comment by danskarda** [comment]
Can we fix it in clojurescript-build or does it mean that
clojurescript-build is obsolete?

**Comment by bhauman** [comment]
I was hoping you guys could run off of a local fix until I got the next
figwheel release ready.

I will take a PR for clojurescript-build and cut an incremental release and
you guys can use that for now.

It will probably be the last release of clojurescript-build as I will be
deprecating it.

**Comment by bhauman** [comment]
Actually after looking at it, this is all wrong I'm worry.

No PR is needed

**Comment by bhauman** [comment]
I think you guys should run a forked build of clojurescript-build in the
short term.

The next figwheel will have this under control.  I will be cutting a
snapshot release of the next figwheel soonish.

**Comment by bhauman** [comment]
Alright I have released 0.5.0-SNAPSHOT 

Please see the reference to `:reload-clj-files` configuration here:
https://github.com/bhauman/lein-figwheel#figwheel-server-side-configuration
